algorithms/src/Kokkos_Sort.hpp

```diff
@@ -665,6 +665,231 @@ void sort(ViewType view, size_t const begin, size_t const end) {
   exec.fence("Kokkos::Sort: fence after sorting");
 }
 
+namespace Impl {
+template <typename Key>
+struct LessThan {
+  KOKKOS_INLINE_FUNCTION bool operator()(const Key lhs, const Key rhs) const {
+    return lhs < rhs;
+  }
+};
+
+// When just doing sort (not sort_by_key), use nullptr_t for ValueViewType.
+// TODO: when nested policies get refactored, this won't need separate team and
+// thread versions.
```

Comment:
I should have thought of that before🤦‍♂️
However, there is still the issue of detecting whether a ``t.team_barrier()`` is needed after each parallel_for (true for team-level, false for thread-level). Not sure I can use ``is_same`` on the range's type, because that seems to be backend and index type specific, like for serial it was ``Kokkos::Impl::TeamThreadRangeBoundariesStruct<unsigned long, Kokkos::Impl::HostThreadTeamMember<Kokkos::Serial> >``. Is there another way to check team vs. thread?

Reply:
Nvm, i think i figured out how to do it